tests/test_general.py

```diff
@@ -844,6 +844,24 @@ def test_error_analytical(self):
     #     self.assertAlmostEqual(var_a_exact**0.5, fit_result.params.a_stdev, 6)
     #     self.assertAlmostEqual(var_b_exact**0.5, fit_result.params.b_stdev, 6)
 
+    def test_fixed_parameters(self):
+        """
+        Make sure fixed parameters don't change on fitting
+        """
+        xdata = np.arange(100)
+        ydata = np.arange(100)
+        
+        a, b, c, d = parameters('a, b, c, d')
+        x, y = variables('x, y')
+        
+        c.value = 4.0
+        c.fixed = True
+        
+        model_dict = {y: a * exp(-(x - b)**2 / (2 * c**2)) + d}
+        fit = Fit(model_dict, x=xdata, y=ydata)
+        fit_result = fit.execute()
+        self.assertEqual(4.0, fit_result.params['c'])
```

Comment:
I'm not sure they should be the same, to be honest. I would argue that setting a variable as fixed it's just not allowed to change during the fitting (i.e. treat it as a constraint). This means that the models where there's a parameter less is fundamentally different.

Furthermore, I did not touch anything around covariance matrices, or in fit.py, so even if we leave that for now it doesn't change existing behaviour.

Reply:
Actually you are right, I was still thinking about 0.3.x, where the covariance matrix returned by the minimizer is used. Since we now redetermine it in a uniform way my critique does not apply. Nice.